### ISSUE TITLE
Feature request: Should -BeEqualTo

### ISSUE THREAD

**Issue by iRon7** [pester/Pester#968]
I have written a [`ConvertTo-PSON` (PowerShell Object Notation)](https://powersnippets.com/convertto-pson/) cmdlet which I am about to upload to github but I want to add some pester tests before I do.
There are lot of situation were I would like to tests both the value (`Should -Be`) *and* its type (`Should -BeOfType`). With requires a lot of statements likes:

```
$Char | Should -Be 'A' ; $Char | Should -BeOfType [Char]
$Long | Should -Be 65 ; $Long | Should -BeOfType [Long]
```

I would like to be able to simplify this to something like:

```
$Char | Should -BeEqualTo ([Char]'A')
$Long | Should -BeEqualTo ([Long]65)
```

Basically: [`[Object]::Equals($ActualValue, $Value)`](https://stackoverflow.com/a/44594861/1701026)  
In this idea, `Should -BeEqualTo` (or `Should -BeIdenticalTo`) should produce an error when either the value  (`Should -Be`) or the type (`Should -BeOfType`) is not equal. Meaning the following two pester statements should both produce a pester error:

```
[Long]64 | Should -BeEqualTo ([Long]65)
[Int]65 | Should -BeEqualTo ([Long]65)
```

**Comment by nohwnd** [comment]
Thank you for your contribution, but I am sorry, I don't think many users would find this useful, because such comparison is very rarely done, and it would be better to split it in two different tests. Naming it `EqualTo` would also make it very confusing. You can still register your own operator to `Should` if you find it useful in your tests, or you can write a simple assertion yourself that does the check and does not use Should to simplify your tests.
